woocommerce/woocommerce-android
# Product Creation with AI: Analytics for product name generation

Part of: #9843


### Description


Add the following events related to product name generation with AI:

| Event Name | Description | Properties |
| --- | --- | --- |
| _product_name_ai_entry_point_tapped | When the entry point to create product name with AI is tapped. | has_input_name: true \| false, source: product_creation_ai |
| _product_name_ai_generate_button_tapped | When the button to generate product name with AI is tapped. | is_retry: true \| false |
| _product_name_ai_copy_button_tapped | When the button to copy the generated product name is tapped. |  |
| _product_name_ai_apply_button_tapped | When the button to apply the generated product name is tapped. |  |
| _product_name_ai_generation_success | When generating a product name with AI succeeds. |  |
| _product_name_ai_generation_failed | When generating a product name with AI fails. | Error detail properties |


### Testing instructions


Please check logcat for the tracking above, by doing these actions:

1. Start product creation with AI and tap "Suggest a name" for `_product_name_ai_entry_point_tapped`
2. Enter some value and generate a name for `_product_name_ai_generate_button_tapped` with `is_retry: false`, and `_product_name_ai_generation_success` as well
4. Once generated, use "Regenerate" button for `_product_name_ai_generate_button_tapped` with `is_retry: true`
5. Try copy button for `_product_name_ai_copy_button_tapped`
6. Try apply button for `_product_name_ai_apply_button_tapped`
7. Start over from point 2, turn off internet before generating, for `_product_name_ai_generation_failed`
